manifests/init.pp

```diff
@@ -1,22 +1,83 @@
-# Class: sftp_jail
-# ===========================
+# @summary sftp_jail class
 #
-# Full description of class sftp_jail here.
+# @param chroot_base
+#   The directories path, which holds all jails.
 #
-# Parameters
-# ----------
+# @param sftp_users
+#   See `Sftp_jail::Sftp_Users` and `Sftp_jail::Sftp_User` types.
 #
-# * `sample parameter`
-#   Explanation of what this parameter affects and what it defaults to.
-#   e.g. "Specify one or more upstream ntp servers as an array."
+# @param manage_users
+#   Should this module manage the user resources?
+#
+# @param sub_dirs
+#   Default subdirectories to ensure in every users home. Can be overwritten for
+#   each user seperatly.
+#
+# @param merge_subdirs
+#   Merge sub_dirs with the default sub_dirs?
+#
+# @param admin_sub_dirs
+#   Default subdirectories to ensure in every admins home. Can be overwritten for
+#   each admin seperatly.
+#
+# @param password_authentication
+#   Password Authentication setting for SFTP users and Admins.
 #
 class sftp_jail (
-  Stdlib::Absolutepath $chroot_base = '/chroot',
+  Sftp_jail::Sftp_Users    $sftp_users              = {},
+  Stdlib::Absolutepath     $chroot_base             = '/chroot',
+  Boolean                  $manage_users            = false,
+  Sftp_jail::Sub_dirs      $sub_dirs                = [],
+  Boolean                  $merge_subdirs           = false,
+  Sftp_jail::Sub_dirs      $admin_sub_dirs          = [],
+  Sftp_jail::Pw_auth       $password_authentication = {},
 ) {
+  $sftp_admins = $sftp_users.filter |$k, $v| { $v['sftp_admin'] }
+
   file { $chroot_base:
     ensure => 'directory',
     owner  => 'root',
     group  => 'root',
     mode   => '0755',
   }
+
+  unless $sftp_admins.empty {
+    file { "${chroot_base}/home":
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0755',
+      require => File[$chroot_base],
```

Comment:
Since Puppet 2.7 this should be done automatically.

Reply:
Ah, I will drop it then.

Reply:
I tested: it creates in with the correct permissions and the correct owner, but it doesn't enforce it if the directory exists with wrong permissions or ownership. The ownership and permissions have to be correct, that's important for the modules functionality.
I can drop it, but I would not.

Reply:
To be clear, I was commenting on *just* the `require` line. Not the entire file resource. So:
```suggestion
```

Reply:
droped them to: https://github.com/voxpupuli/puppet-sftp_jail/pull/69/commits/66a1d80cac3ed00c93ead2bee1338d2d95cd220e